fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -105,9 +116,11 @@ def signed_weights(self, lambda_vec):
         :param lambda_vec: The vector of Lagrange multipliers indexed by `index`
         :type lambda_vec: :class:`pandas:pandas.Series`
         """
-        lambda_signed = lambda_vec["+"] - lambda_vec["-"]
-        adjust = lambda_signed.sum(level=_EVENT) / self.prob_event \
-            - lambda_signed / self.prob_group_event
+        lambda_event = (lambda_vec["+"] - self.ratio * lambda_vec["-"]) \
```

Comment:
These already look correct / up-to-date with the updates to the derivations I discussed with @MiroDudik today. Looks like you're way ahead of me :-)

Reply:
This is not quite right--we have a hidden dependence on `self.ratio` which is being initialized in subclasses of `ConditionalSelectionRate`. Move it to the constructor of `ConditionalSelectionRate` as @romanlutz suggests below.